frontend/src/components/GeoSelection/index.js

```diff
@@ -294,6 +306,18 @@ export default class GeoSelection extends React.PureComponent {
             flatLocations,
         } = this.state;
 
+        if (flatLocations.length === 0) {
```

Comment:
What if length of the flatLocations is meant to be 0
Use some other variable to identify if it is loading.